### PR TITLE
feat(editor): validate key

### LINKED ISSUE #58: Prevent invalid `key` from being entered
The properties panel has no built-in validation. It should prevent any invalid key (duplicates as well as empty ones).

yLvF9xLRjA

### Reproduce

1. Set the `key` of a field
2. Set the `key` of another field to the same value

### Expected Behavior

1. Duplicate `key`s are prevented
2. Empty `key`s are prevented
3. :arrow_right: All diagrams that are configurable via the `form-js` editor shall be valid, i.e. importable again in editor and viewer instances.

### PR DESCRIPTION
**Pull request by philippfromme**
o2WPRVVqCa

---

Closes #58
Related to https://github.com/camunda/camunda-modeler/issues/2358

### CODE REVIEW

File: packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js [hunk 1]
```diff
@@ -1,19 +1,49 @@
...
   } = props;
 
+  const formFieldRegistry = useService('formFieldRegistry');
+
+  const validate = (value) => {
+    if (isUndefined(value) || !value.length) {
+      return 'Key must not be empty.';
+    }
+
+    if (/\s/.test(value)) {
```
Comment (nikku) [inline packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js:20]: As the context is given we could consider to drop _Key_ as a work (i.e. simplify to _Must not be empty_.

File: packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js [hunk 2]
```diff
@@ -1,19 +1,49 @@
...
+      return 'Key must not contain spaces.';
+    }
+
+    if (!isUnique(value, field, formFieldRegistry)) {
```
Comment (nikku) [inline packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js:24]: Potential to drop `Key` (context already given)

File: packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js [hunk 3]
```diff
@@ -1,19 +1,49 @@
...
   const {
     editField,
     field
   } = props;
 
+  const formFieldRegistry = useService('formFieldRegistry');
+
+  const validate = (value) => {
+    if (isUndefined(value) || !value.length) {
+      return 'Key must not be empty.';
```
Comment (nikku) [inline packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js:17]: Potential to drop `Key` (context already given).

**Review by nikku — COMMENTED** [review]
Works as expected. 

I suggest to color the input focus indicator redish on error (cf. how bootstrap and others do it).

**Comment by philippfromme** [comment]
> I suggest to color the input focus indicator redish on error (cf. how bootstrap and others do it).

I've applied the same styles as in the new properties panel (cf. https://github.com/bpmn-io/properties-panel/pull/75) which applied styles according to this sketch: https://app.zeplin.io/project/5e149dc7ef8eaa51b52af327/screen/60d9619fbeb59812106abf00

I don't like the way it looks, either. What should we do?

File: packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js (hunk 1 above)
Comment (philippfromme) [reply packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js:20]: I've used the same wording as in the new properties panel (cf. https://github.com/bpmn-io/properties-panel/pull/75). What should we do?

Comment (nikku) [reply packages/form-js-editor/src/render/components/properties-panel/entries/KeyEntry.js:20]: Well, [my feedback](https://github.com/bpmn-io/form-js/pull/127#discussion_r667529449) applies for that one, too. (CC @pinussilvestrus, CC @andreasgeier).

Good to have it consistent to start with I guess.

**Comment by nikku** [comment]
> I don't like the way it looks, either. What should we do?

Discuss it with @andreasgeier. Fix it in the original properties panel (add error focus color) and port it to this place, too.

**Comment by philippfromme** [comment]
> Discuss it with @andreasgeier. Fix it in the original properties panel (add error focus color) and port it to this place, too.

Will be discussed today with @andreasgeier.

**Comment by pinussilvestrus** [comment]
I also dicussed it with @andreasgeier, we will update it inside `properties-panel` as well 👍

**Comment by philippfromme** [comment]
@nikku I've adjusted the error messages and the styling as you suggested and as discussed with @andreasgeier. Should be ready now.

**Review by nikku — APPROVED** [review]

**Pull request merged**